analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/descriptions/CommentSummaryDescription.java

```diff
@@ -5,6 +5,15 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import java.util.Map;
 
+/**
+ * Description for a JavaDoc comment.
+ *
+ * @param remarks    The first sentence of the JavaDoc comment?
+ * @param returns    The text in the '@return' tag.
+ * @param summary    The second sentence of the JavaDoc comment?
+ * @param params     A mapping of parameter names to their text, from the '@param' tags.
+ * @param typeParams A mapping of type parameter names to their text, from ?
```

Comment:
The remakrs contains the first sentence in the  javadoc leading to the first "." .
The summary contains all the other sentences after the first "." .
Typeparams also uses @param to get the data usually this data in expected to be inside "< >".

Reply:
![image](https://github.com/Ali-chakaroun/ISEP-LivingDocumentation/assets/58431892/bc534537-66f7-439e-b67b-2dcbeacf7f63)

This only saves the second sentence to the summary, right?

Furthermore, shouldn't the summary contain the first sentence, and remarks the rest? 
"The <remarks> tag is used to add information about a type or a type member, supplementing the information specified with <summary>. This tag may include more lengthy explanations."  from https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags#remarks
Or maybe even better, put everything in summary, as we have no clues to the content of the text, thus cannot say it qualifies as a remark.

Reply:
The `split`'s has a limit of 2, so `sentences[]` is either empty, or 1 element long or 2 elements long, so the summary is the first sentence and the remarks are the rest.

The first sentence being special is per [here](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html):

> First Sentence: The first sentence of each doc comment should be a summary sentence, containing a concise but complete description of the API item.

Reply:
Ahh that makes sense, thanks for explaining! Then I suppose we can manipulate the remarks tag for the rest, even if it is a bit hacky. So both me and Ali got the order switched around, which confused me. Thanks!